diff --git a/http2/server.go b/http2/server.go
index ce2e8b4..6477501 100644
--- a/http2/server.go
+++ b/http2/server.go
@@ -46,6 +46,7 @@ import (
 	"strings"
 	"sync"
 	"time"
+	_ "unsafe"
 
 	"golang.org/x/net/http/httpguts"
 	"golang.org/x/net/http2/hpack"
@@ -2596,6 +2597,9 @@ func (rws *responseWriterState) declareTrailer(k string) {
 	}
 }
 
+//go:linkname appendTime net/http.appendTime
+func appendTime([]byte, time.Time) []byte
+
 // writeChunk writes chunks from the bufio.Writer. But because
 // bufio.Writer may bypass its chunking, sometimes p may be
 // arbitrarily large.
@@ -2638,7 +2642,7 @@ func (rws *responseWriterState) writeChunk(p []byte) (n int, err error) {
 		var date string
 		if _, ok := rws.snapHeader["Date"]; !ok {
 			// TODO(bradfitz): be faster here, like net/http? measure.
-			date = rws.conn.srv.now().UTC().Format(http.TimeFormat)
+			date = string(appendTime(make([]byte, 0, len(http.TimeFormat)), rws.conn.srv.now().UTC()))
 		}
 
 		for _, v := range rws.snapHeader["Trailer"] {
