diff --git a/http2/server.go b/http2/server.go
index 6c349f3..612d4f4 100644
--- a/http2/server.go
+++ b/http2/server.go
@@ -2683,7 +2683,7 @@
 		var date string
 		if _, ok := rws.snapHeader["Date"]; !ok {
 			// TODO(bradfitz): be faster here, like net/http? measure.
-			date = rws.conn.srv.now().UTC().Format(http.TimeFormat)
+			date = string(appendTime(make([]byte, 0, len(http.TimeFormat)), rws.conn.srv.now().UTC()))
 		}
 
 		for _, v := range rws.snapHeader["Trailer"] {
diff --git a/http2/server_time.go b/http2/server_time.go
new file mode 100644
index 0000000..fa7248b
--- /dev/null
+++ b/http2/server_time.go
@@ -0,0 +1,26 @@
+package http2
+
+import (
+	"time"
+)
+
+func appendTime(b []byte, t time.Time) []byte {
+	const days = "SunMonTueWedThuFriSat"
+	const months = "JanFebMarAprMayJunJulAugSepOctNovDec"
+
+	t = t.UTC()
+	yy, mm, dd := t.Date()
+	hh, mn, ss := t.Clock()
+	day := days[3*t.Weekday():]
+	mon := months[3*(mm-1):]
+
+	return append(b,
+		day[0], day[1], day[2], ',', ' ',
+		byte('0'+dd/10), byte('0'+dd%10), ' ',
+		mon[0], mon[1], mon[2], ' ',
+		byte('0'+yy/1000), byte('0'+(yy/100)%10), byte('0'+(yy/10)%10), byte('0'+yy%10), ' ',
+		byte('0'+hh/10), byte('0'+hh%10), ':',
+		byte('0'+mn/10), byte('0'+mn%10), ':',
+		byte('0'+ss/10), byte('0'+ss%10), ' ',
+		'G', 'M', 'T')
+}
